lib/mongoid/criterion/scrollable.rb

```diff
@@ -17,11 +17,10 @@ def scroll(cursor = nil, &_block)
         field = criteria.klass.fields[scroll_field.to_s]
         cursor_options = { field_type: type_from_field(field), field_name: scroll_field, direction: scroll_direction }
         cursor = cursor.is_a?(Mongoid::Scroll::Cursor) ? cursor : Mongoid::Scroll::Cursor.new(cursor, cursor_options)
+        criteria.selector = { '$and' => [criteria.selector, cursor.criteria] }
         # scroll
         if block_given?
-          cursor_criteria = criteria.dup
```

Comment:
I think this was here for a reason. Every critiera operation returns a new criteria. Looks like we're missing a spec too because nothing broke after this was removed. At the very least maybe put it back and open a new issue?

Reply:
Ah, missed that. I'll try my hand at making a spec for it!

Reply:
I've added the following spec, but it passes, am I testing for the right thing?
```
it 'does not change original criteria' do
  criteria = Feed::Item.where(:a_time.gt => Time.new(2013, 7, 22, 1, 2, 3))
  criteria.limit(2).scroll
  expect(criteria).to eq criteria
  cursor = nil
  criteria.limit(2).scroll do |record, next_cursor|
    records << record
    cursor = next_cursor
  end
  expect(criteria).to eq criteria
end
```

Reply:
OK here's my first stab at a spec. It passes, so I'm wondering if I'm understanding the issue correctly:

```
it 'does not change original criteria' do
  criteria = Feed::Item.where(:a_time.gt => Time.new(2013, 7, 22, 1, 2, 3))
  original_criteria = criteria.dup
  criteria.limit(2).scroll
  expect(criteria).to eq original_criteria
  cursor = nil
  criteria.limit(2).scroll(cursor) do |_record, next_cursor|
    cursor = next_cursor
  end
  expect(criteria).to eq original_criteria
end
```

Reply:
If you don't `scroll` the second time but reuse the criteria for something else you should be able to get different results (because we tacked an `$or` on it. Remove that dup in the current code and iterate until you have a failing spec.

If you can't figure it out I am OK with an issue too and we can try separately.

Reply:
I'm not 100% sure what the problem is, so I think it's better to open an issue and dive into it there. I'll put back the original `dup` call and we discuss in the new issue. What should the topic be of the new issue?

Btw I also have some super simple refactoring proposals, just to make the code a bit more readable. I can make a separate merge later on though, if that is of interest?